Feature/aws lambda

shellygamlielly opened the pull request.

shellygamlielly committed:
install aws-serverless-express and initialize the NestJS application with AWS Lambda support

shellygamlielly committed:
serverlessExpress, webpack (boundle all) ,serverless offline for dev, mongo is used once on startup,

shellygamlielly committed:
if development mode the handler returns null and local bootstrap is called

shellygamlielly committed:
add AuthInterceptor to controller

shellygamlielly committed:
Get spotify id given access token and use the user id to allow sending answer from the server.

shellygamlielly committed:
enable cors and add dev env variable

shellygamlielly committed:
Merge branch 'master' into feature/aws-lambda

shellygamlielly committed:
github action- hello world

shellygamlielly committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/aws-lambda' into feature/aws-lambda

shellygamlielly committed:
change directory

shellygamlielly committed:
update the actions/checkout@v2 to use Node.js 20:

shellygamlielly committed:
print node version

shellygamlielly committed:
install npm and webpack

shellygamlielly committed:
npm run webpack

shellygamlielly committed:
print main.js line count

shellygamlielly committed:
Create deployment zip file. Install and configure AWS CL. Deploy to Lambda

shellygamlielly committed:
add aws-region

shellygamlielly committed:
armani

shellygamlielly committed:
ara

shellygamlielly committed:
branch is master

ramiloif commented on .github/workflows/lambda-deploy.yml:
```diff
@@ -0,0 +1,49 @@
+      - name: Build NestJS application with Webpack
+        run: npm run webpack
+
+      - name: Print main.js Line Count
```
Remove this step

shellygamlielly replied:
only print not webpack right?

ramiloif replied:
#Yea just the print you done for debug

ramiloif commented on .github/workflows/lambda-deploy.yml:
```diff
@@ -0,0 +1,49 @@
+        run: |
+          zip deployment.zip dist/main.js
+
+      - name: ppp
```
remove this step

ramiloif commented on serverless.yml:
```diff
@@ -0,0 +1,19 @@
+service: serverless-tunity
```
Thus yaml is only for testing locally ?

shellygamlielly replied:
Since I have configured the Lambda function settings, including the handler, in the AWS Lambda console, that becomes the primary source of configuration during production deployments. The serverless.yml file can still be valuable for local testing with the "serverless-offline" plugin. In the production deployment process managed by GitHub Actions and aws lambda update-function-code, the serverless.yml file may have a limited role since it is already managed in the AWS Lambda console.

ramiloif commented on .github/workflows/lambda-deploy.yml:
```diff
@@ -0,0 +1,49 @@
+
+      - name: Create deployment zip file
+        run: |
+          zip deployment.zip dist/main.js
```
<img width="188" alt="image" src="https://github.com/shellygamlielly/tunity-server/assets/15690633/9a3a9884-1d90-454c-8cc6-ff39202157ae">
Thats probably correct since the config is `dist/main.handler`

shellygamlielly replied:
if you are correct it should be : " zip **-r** deployment.zip dist/main.js" this will create it in a "dist" directory. But if it worked for you we might not change it.

ramiloif replied:
Yea leave as is

ramiloif commented on src/interceptors/auth.interceptors.ts:
```diff
@@ -0,0 +1,50 @@
+    context: ExecutionContext,
+    handler: CallHandler,
+  ): Promise<Observable<any>> {
+    const req = context.switchToHttp().getRequest();
```
you can add here in the top of the file 

```
  if (true)
  return handler.handle();
```
Just to be able to merge this PR withot breaking the app :)

shellygamlielly replied:
Notice that iIdefined : "req.userId = await this.userService.getUserBySpotifyId(spotifyUserId)", so for example if you want to create a playlist :  @Post('')
  async createPlaylist(@Body() playlistName: string, @Req() req: any) {
    return await this.playlistService.createPlaylist(req.userId, playlistName);
  }. And if i dont set the userId, a playlist cannot be created.

shellygamlielly committed:
cleanup

shellygamlielly committed:
Fix interceptor: Added AuthInterceptor to providers with @Injectable().
Updated createPlaylist method to correctly extract the name of the playlist from the request body.
Listens on port 3000.
